packages/vue/src/components/molecules/SfComponentSelect/SfComponentSelect.vue

```diff
@@ -0,0 +1,285 @@
+import SfOverlay from "../../atoms/SfOverlay/SfOverlay.vue";
+import { focus } from "../../../utilities/directives";
+import { clickOutside } from "../../../utilities/directives";
+import Vue from "vue";
+Vue.component("SfComponentSelectOption", SfComponentSelectOption);
+export default {
+  name: "SfComponentSelect",
+  directives: { focus, clickOutside },
+  components: {
+    SfButton,
+    SfChevron,
+    SfOverlay,
+  },
+  model: {
+    prop: "selected",
+    event: "change",
+  },
+  props: {
+    /**
+     * Select field label
+     */
+    label: {
+      type: String,
+      default: "",
+    },
+    /**
+     * Selected item value
+     */
+    selected: {
+      type: [String, Number, Object],
+      default: "",
+    },
+    /**
+     * Dropdown list size
+     */
+    size: {
+      type: Number,
+      default: 5,
+    },
+    /**
+     * Required attribute
+     */
+    required: {
+      type: Boolean,
+      default: false,
+    },
+    /**
+     * Validate value of form select
+     */
+    valid: {
+      type: Boolean,
+      default: true,
+    },
+    /**
+     * Disabled status of form select
+     */
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    /**
+     * Error message value of form select. It will be appeared if `valid` is `true`.
+     */
+    errorMessage: {
+      type: String,
+      default: "This field is not correct.",
+    },
+    /**
+     * If true clicking outside will not dismiss the select
+     */
+    persistent: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      open: false,
+      options: [],
+      indexes: {},
+      optionHeight: 0,
+      focusedOption: "",
+    };
+  },
+  computed: {
+    index: {
+      get() {
+        const stringified = this.indexes[JSON.stringify(this.selected)];
+        if (typeof stringified === "undefined") {
+          return -1;
+        }
+        return stringified;
+      },
+      set(index) {
+        this.focusedOption = this.options[index].value;
+        this.$emit("change", this.options[index].value);
+      },
+    },
+    html() {
+      if (this.index < 0) return;
+      return this.options[this.index].html;
+    },
+    maxHeight() {
+      if (!this.options.length) return;
+      return `${this.optionHeight * this.options.length}px`;
+    },
+    isActive() {
+      return this.open;
+    },
+    isSelected() {
+      return this.selected;
+    },
+  },
+  watch: {
+    open: {
+      immediate: true,
+      handler: function (visible) {
+        if (visible) {
+          this.$nextTick(() => {
+            if (this.$slots.default) {
+              this.optionHeight = this.$slots.default[0].elm.offsetHeight;
+            }
+          });
+        }
+      },
+    },
+  },
+  created: function () {},
+  mounted: function () {
+    this.addOptionsAndIndexes();
+  },
+  updated() {
+    if (this.$slots.default) {
+      if (this.$slots.default.length > this.options.length) {
```

Comment:
```suggestion
      if (this.$slots.default && this.$slots.default.length > this.options.length) {
```
Maybe one line instead of two, wdyt?

Reply:
Yeah, why not? :)